web/src/components/tasks/business-formation/business/MainBusiness.tsx

```diff
@@ -45,26 +47,34 @@ export const MainBusiness = (): ReactElement => {
         </WithErrorBar>
       </WithErrorBar>
       {isForeign && (
-        <WithErrorBar
-          hasError={doSomeFieldsHaveError(["foreignStateOfFormation", "foreignDateOfFormation"])}
-          type="DESKTOP-ONLY"
-          className="grid-row tablet:grid-gap-1"
-        >
-          <WithErrorBar
-            hasError={doesFieldHaveError("foreignStateOfFormation")}
-            type="MOBILE-ONLY"
-            className="tablet:grid-col-6"
-          >
-            <ForeignStateOfFormation />
-          </WithErrorBar>
+        <>
           <WithErrorBar
-            hasError={doesFieldHaveError("foreignDateOfFormation")}
-            type="MOBILE-ONLY"
-            className="tablet:grid-col-6"
+            hasError={doSomeFieldsHaveError(["foreignStateOfFormation", "foreignDateOfFormation"])}
+            type="DESKTOP-ONLY"
+            className="grid-row tablet:grid-gap-1"
           >
-            <FormationDate fieldName="foreignDateOfFormation" />
+            <WithErrorBar
+              hasError={doesFieldHaveError("foreignStateOfFormation")}
+              type="MOBILE-ONLY"
+              className="tablet:grid-col-6"
+            >
+              <ForeignStateOfFormation />
+            </WithErrorBar>
+            <WithErrorBar
+              hasError={doesFieldHaveError("foreignDateOfFormation")}
+              type="MOBILE-ONLY"
+              className="tablet:grid-col-6"
+            >
+              <FormationDate fieldName="foreignDateOfFormation" />
+            </WithErrorBar>
           </WithErrorBar>
-        </WithErrorBar>
+          {state.formationFormData.legalType === "foreign-c-corporation" && (
+            <PracticesLaw hasError={doesFieldHaveError("willPracticeLaw")} />
+          )}
+          {state.formationFormData.legalType === "foreign-c-corporation" && (
+            <ForeignCertificate hasError={doesFieldHaveError("foreignGoodStandingFile")} />
+          )}
```

Comment:
These could both go within the same logic check instead of repeating it?

Reply:
Also - I'm noticing that the other components like `FormationDate` and `ForeignStateOfFormation` are independent components, and then are wrapped in the WithErrorBar here in the parent-display layout. Whereas you included the WithErrorBar wrapper internal to the ForeignCertificate. 

I'd argue that makes more sense to keep it external because the error bar relies on assumptions about layout. Specifically the MOBILE/DESKTOP view. If the file input were side-by-side with another component, the chosen "ALWAYS" prop would have to change, and that would be hard to track internally. Hence my argument for keeping the ErrorBar at the top level like we do for the other components here.

Reply:
That reasoning makes sense, but I'm leaving `WithErrorBar` nested. I see this pattern in other components like `BusinessFormationTextField`. These components really only exist to be a part of this form and aren't affected by the surrounding items. I think it just eliminates a little clutter from `MainBusiness`.